prime-angular-frontend/src/app/modules/enrolment/pages/regulatory/regulatory.component.ts

```diff
@@ -84,6 +85,11 @@ export class RegulatoryComponent extends BaseEnrolmentProfilePage implements OnI
     this.routeTo(EnrolmentRoutes.CARE_SETTING);
   }
 
+  public shouldRemoveRemoteAccess(): boolean {
+    return this.hasRemoteAccess && !this.isInitialEnrolment &&
```

Comment:
Feels like the best solution might be a `valueChanges` listener in `initForm`, which would mean you could just use the `*ngIf` directive with `!this.hasRemoteAccess && !this.isInitialEnrolment` and not have to invoke a method in the controller when it can be avoided:

```
this.form.valueChanges
  .subscribe(() => this.hasRemoteAccess = this.canRequestRemoteAccess());
```

The advantage of doing it this way means: 
1. It only checks when an update occurs to the form
2. Preserves the use of the member variable `hasRemoteAccess`, which was only being used on initialization until the first form update, and 
3. Removes a method that would be invoked by the template 100s of times per minute depending on how the user interacts with the application as it would keep checking for changes that can only occur when the form changes.  You can test this out by logging the form updates in the subscribe vs the `shouldRemoveRemoteAccess` method

Reply:
Good thinking. However, the *ngIf checks and display the warning message when enrollee currently has remote access but changes to a licence type which is no longer eligible. If this.hasRemoteAccess is subscribed to the form.valuechanges,  and init to be false, which leads to the warning messages to be displayed. This is not expected for an enrollee currently without remote access.
This idea is to initialize hasRemoteAccess and it indicates the current status of remote access before any changes. Then we determine if the enrollee has changes in license type in the form by calling canRequestRemoteAccess. The shouldRemoveRemoteAccess is just the final logical check. I feel a bit cleaner having a descriptive function than writing the logic directly like *ngIf="this.hasRemoteAccess && !this.isInitialEnrolment && !this.canRequestRemoteAcccess()" If there is a better solution, let me know 😊

Reply:
Agreed with the extra firing of the method. Changing it subscribe to valuchanges.

> 3. Removes a method that would be invoked by the template 100s of times per minute depending on how the user interacts with the application as it would keep checking for changes that can only occur when the form changes.  You can test this out by logging the form updates in the subscribe vs the `shouldRemoveRemoteAccess` method

Reply:
@johnnyl6 thanks, so instead of using a member variable that only has a singular purpose of indicating an initial value I would use a closure.  To keep with self documenting code we could name it `initialRemoteAccess`, and also rename `shouldRemoveRemoteAccess` to `cannotRequestRemoteAccess` which aligns a better with its compliment `canRequestRemoteAccess`:

```
public ngOnInit() {
  this.createFormInstance();
  this.patchForm().subscribe(() => this.initForm());

  const initialRemoteAccess = this.canRequestRemoteAccess();

  this.form.valueChanges
    .pipe(map((_) => initialRemoteAccess))
    .subscribe((initialRemoteAccess: boolean) =>
      this.cannotRequestRemoteAccess = initialRemoteAccess && !this.canRequestRemoteAccess() && !this.isInitialEnrolment
    );
}
```

The only other change I would make is initializing `this.cannotRequestRemoteAccess` in the `constructor` instead of `initForm`.